client/my-sites/email/email-management/email-home.tsx

```diff
@@ -131,24 +140,31 @@ const EmailHome = ( props: EmailManagementHomeProps ) => {
 		if ( ! domainHasEmail( selectedDomain ) ) {
 			return (
 				<EmailProvidersStackedComparisonPage
+					className={ clsx( { 'context-all-domain-management': isAllDomainManagementContext } ) }
 					comparisonContext="email-home-selected-domain"
 					selectedDomainName={ selectedDomainName }
 					selectedEmailProviderSlug={ selectedEmailProviderSlug }
 					selectedIntervalLength={ selectedIntervalLength }
 					source={ source }
+					hideNavigation={ isAllDomainManagementContext }
 				/>
 			);
 		}
 
 		return (
-			<ContentWithHeader>
+			<ContentWithHeader
+				className={ clsx( { 'context-all-domain-management': isAllDomainManagementContext } ) }
+			>
 				<EmailPlan
 					domain={ selectedDomain }
 					// When users have a single domain with email, they are auto-redirected from the
 					// `/email/:site_slug` page to `/email/:domain/manage/:site_slug`. That's why
 					// we also hide the back button, to avoid scenarios where clicking "Back"
 					// redirects users to the same page as they are currently on.
-					hideHeaderCake={ isSingleDomainThatHasEmail }
+					hideHeaderCake={ isAllDomainManagementContext || isSingleDomainThatHasEmail }
+					hideHeader={ isAllDomainManagementContext }
+					hidePlanActions={ isAllDomainManagementContext }
+					hideMailPoetUpsell={ isAllDomainManagementContext }
```

Comment:
Do you think it might make sense to pass a single `context` prop instead and have `EmailPlan` determine internally which components to hide based on that prop..?

Reply:
I didn't want to introduce "context" in the presentational component; instead, the parent component would have to care about the context type. 

Once the time comes, we will remove the context logic and keep only one. In that case, it will be easy to maintain/update it.